packages/components/src/icon-button/index.js

```diff
@@ -23,7 +23,7 @@ class IconButton extends Component {
 		const { icon, children, label, className, tooltip, shortcut, labelPosition, ...additionalProps } = this.props;
 		const { 'aria-pressed': ariaPressed } = this.props;
 		const classes = classnames( 'components-icon-button', className, {
-			'has-text': children,
+			'has-text': children && children.length > 2,
```

Comment:
`children` is an opaque object type, and `length` is not intended to be reliable.

See also: https://reactjs.org/docs/react-api.html#reactchildren
Specifically: https://reactjs.org/docs/react-api.html#reactchildrencount

```js
import { Children } from '@wordpress/element';

// ...

'has-text': Children.count( children ) > 2
```

Reply:
Thanks for the code improvement suggestion, much appreciated. Though counting the children to be more than 2 on its own feels hacky — do you have any suggestions for how to make sure that the "has-text" returns true when the IconButton has `[Icon] Label`, but false in cases of `[Icon] [Icon]` in a more reliable way?